test/sequential/test-module-loading.js

```diff
@@ -247,12 +247,12 @@ assert.deepEqual(children, {
 assert.throws(function() {
   console.error('require non-string');
   require({ foo: 'bar' });
-}, 'path must be a string or Buffer');
+}, /path must be a string/);
```

Comment:
@thefourtheye Just in case there's a misunderstanding: This is not more lenient than the current test, although it appears more lenient thanks to the unfortunate way the `assert.throws()` API works.

The way it is without this PR, any throw will result in the test passing. The string (`'path must be a string or Buffer'`) is not checked at all. Instead, that string is used as the message provided by the `AssertionError` that fires if the function does not throw.  Fun, right?

Changing it to a RegExp the way this PR does means that it will be used to confirm the message in the `Error` thrown by the function, which is probably what was intended in the first place.